### ISSUE TITLE
Unable to run with given nginx configurations

### ISSUE THREAD

**Issue by grantbacon** [mehrvarz/webcall#7]
Very neat project! I was trying to run my own instance behind nginx as a reverse proxy. I followed the steps https://timur.mobi/webcall/install/

Using config.ini as such:

```
httpsPort = 0
hostname = hostname.url
wssPort = 8443
```

nginx config is same as provided on URL above (adjusted to point to proper webroot).

I also checked precedence to ensure it's not caught by any other location handlers... that said I can access the callee page, get assigned an ID but then when I'm taken to https://url/callee/123456789, I get a 404...   `"GET /callee/11727440667 HTTP/2.0" 404 107`

I tried playing with try_files configs to point to full path to callee/index.html and some other things, but all to no avail.

There's nothing suspicious in my error.log, and I do not see activity in the webcall standard output (except registration calls which work fine)

What is it that handles routing URLs ending in numbers beyond the nginx confiugration? Please let me know if I can provide any additional information, would love to get this working. :)

**Comment by mehrvarz** [comment]
> try_files $uri $uri/ /webroot/callee/index.html;

The point of try_files is for nginx to serve the file "$uri", if it exists, otherwise serve "$uri/" (without the file portion), if that exist, and otherwise serve the 3rd option. So if $uri ends in "/callee/nnnn" and no such file exists  under the root folder, the file "/webroot/callee/index.html" is being served to the web browser.

The web browser is not aware of any of that. It has requested "/callee/nnnn" and was given proper markup in return. And when the variable window.location is being evaluated, it is (still) set to "/callee/nnnn". This is how JS is able to use "nnnn" as the WebCall ID.

In short: browser requests nnnn, nginx ignores the ID and serves a static html file. JS code then makes use of the ID when it talks to WebCall server directly.

**Comment by mehrvarz** [comment]
If GET /callee/11727440667 gives you a 404, maybe there is something wrong with your 3rd try_files option.

**Comment by grantbacon** [comment]
So I managed to get success from removing /webroot/ from the beginning of my final try_files argument, and strangely enough that works for me. 

> location /callee {
> root /opt/webcall/webroot/;
> index index.html index.htm;
> try_files $uri $uri/ /callee/index.html;
> }

I'm not sure why it differs from your config.. this is on nginx 1.18.0-6.1 from debian 11 repo.

But all is well on my end. :+1:  


PS: 
Typo  in your [docs ](https://timur.mobi/webcall/install/) you refer to https running on port 80 at the end of section titled: "Using Nginx as a https reverse proxy" (line 354 of the HTML)

Excellent setup docs though, and easy to do. Glad to have it working!

**Comment by mehrvarz** [comment]
Good to know you have it working. Your location /callee config now looks like what I expected when I first looked at this. No idea what is causing the anomaly on my system. I am also running 1.18. Anyways, I fixed my doc to reflect this. I also removed port 80 further down. Thanks.
